.pre-commit-config.yaml

```diff
@@ -30,13 +30,14 @@ repos:
   - id: pydocstyle
     exclude: dags/|(.*/)?test_.*\.py|setup.py
 - repo: https://github.com/pre-commit/mirrors-mypy
-  # Using commit hash corresponding to tag v1.9.0, using the tag directly was resulting in cannot resolve error.
-  rev: '9db9854'
+  rev: v0.991
```

Comment:
While this is a downgrade, I think it's important to keep the pre-commit checks aligned with the CI checks, and unfortunately for the CI checks we're stuck on this ancient version of `mypy` from 2022 due to `pytest-mypy` not receiving any updates in over two years (though there were some comments made toward the end of 2024 about a forthcoming `pytest-mypy` update).

Reply:
Agreed that they should be in sync, but would it make more sense to replace `pytest-mypy` with just an additional command to run `mypy` separately? Is there another advantage to `pytest-mypy` other than saving a command that makes it worthwhile to remain a few years out of date? Maybe moot if we expect a forthcoming update, and either way not suggesting you need to fix it in this PR.

Thanks for finding/fixing this!

Reply:
I think switching to running `mypy` directly would be reasonable, especially if the `pytest-mypy` updates continue to not happen.  However, right now we're using `pytest` to run a [bunch of things through one interface](https://github.com/mozilla/bigquery-etl/blob/16cd568241895c2dc74d0805c27141334ecbb069/script/entrypoint#L31) beyond just `mypy`, so one downside is splitting `mypy` out could make that setup a bit awkward.

@scholtzan what do you think?

Reply:
Yeah, we should run `mypy`, which is also what we're doing in a couple of other projects (like jetstream) already. So there is precedence